src/Lower.cpp

```diff
@@ -174,6 +174,7 @@ Module lower(const vector<Function> &output_funcs,
     debug(2) << "Lowering after sliding window:\n" << s << '\n';
 
     debug(1) << "Performing allocation bounds inference...\n";
+    s = simplify_correlated_differences(s);
```

Comment:
I'd really like all passes to have their own debug messages here.

Reply:
Hmm... I guess we already lost that property to the lower function. It's probably time to consider adding a little bit of structure here to allow consistent printing of timing and debug info at varying levels of granularity.

Reply:
Done

Reply:
It's already a little heirarchical, in that some of the passes (e.g. partition loops) are internally arranged as a sequence of sub-passes. We could just break out all the passes here.